pkgs/applications/editors/neovim/default.nix

```diff
@@ -89,7 +107,7 @@ in
       substituteInPlace src/nvim/version.c --replace NVIM_VERSION_CFLAGS "";
     '';
     # check that the above patching actually works
-    disallowedReferences = [ stdenv.cc ];
+    disallowedReferences = [ stdenv.cc ] ++ lib.optional (lua != codegenLua) codegenLua;
```

Comment:
```suggestion
    disallowedReferences = [ stdenv.cc ] ++ lib.optional lua.pkgs.isLuaJIT codegenLua;
```

Reply:
I don't see why you think the proposed change would be an improvement.

IMHO the original code fairly clearly showed the intent that, if a different lua was used for code generation than for the runtime, then we don't want to codegen lua to 'leak' into the references. The new notation might save a 'not' and some parentheses, but now to understand what's going on on this line you have to understand that `codegenLua` is different from `lua` if and only if `lua.pkgs.isLuaJIT` is true. That seems less readable/maintainable to me.

Reply:
The condition will be wrong if the passed in lua is overwritten to `` deterministicLuajit.withPackages (ps: with ps; [ mpack lpeg ])``. Also chaining condition on the results of other conditions is bad coding style and makes following the code more complicated. Both condition should be based on the same boolean expression.

Reply:
> The condition will be wrong if the passed in lua is overwritten to  `deterministicLuajit.withPackages (ps: with ps; [ mpack lpeg ])`.

if the passed in lua is overwritten to `deterministicLuajit.withPackages (ps: with ps; [ mpack lpeg ])` then it is expected that there is a runtime reference to that version of Lua, so in that case `disallowedReferences = [ stdenv.cc ] ++ lib.optional (lua != codegenLua) codegenLua;` is correct and `disallowedReferences = [ stdenv.cc ] ++ lib.optional lua.pkgs.isLuaJIT codegenLua;` is wrong.

> chaining condition on the results of other conditions is bad coding style and makes following the code more complicated. Both condition should be based on the same boolean expression.

I disagree, I'd argue in some cases (like here) repeating the same boolean logic in multiple places is bad coding style, and referring to well-named variables captures the intent of the code better and makes it easier to understand, not more complicated.

Reply:
> I'd argue in some cases (like here) repeating the same boolean logic in multiple places is bad coding style

then it should rather be moved to a common variable in a let in